scheduler/src/cook/mesos/api.clj

```diff
@@ -1408,15 +1425,23 @@
 (defn check-jobs-and-groups-exist
   "Check if all of the ::jobs and ::groups in the context are valid."
   [conn ctx]
-  (and (check-jobs-exist conn ctx)
-       (check-groups-exist conn ctx)))
+  (let [[jobs-ok? jobs-ctx' :as jobs-result] (vectorize (check-jobs-exist conn ctx))]
+    (if-not jobs-ok?
+      jobs-result
+      (let [[groups-ok? groups-ctx' :as groups-result] (vectorize (check-groups-exist conn ctx))]
+        (cond
+          (not groups-ok?) groups-result
+          (not (or jobs-ctx' groups-ctx')) true
+          (and jobs-ctx' groups-ctx') [true (combine jobs-ctx' groups-ctx')]
+          :else [true (or jobs-ctx' groups-ctx')])))))
```

Comment:
The logic here became a lot more complex because I didn't want any potential future changes to `check-jobs-exist` or `check-groups-exist` to unexpectedly break this function.